packages/core/src/orchestrator.ts

```diff
@@ -257,22 +258,30 @@ export function createEnvelopOrchestrator<PluginsContext extends DefaultContext>
 
   const customContextFactory: EnvelopContextFnWrapper<(orchestratorCtx?: any) => any, any> = beforeCallbacks.context.length
     ? initialContext => async orchestratorCtx => {
-        const afterCalls: OnContextBuildingHook<any>[] = [];
+        const afterCalls: AfterContextBuildingHook<any>[] = [];
 
         try {
           let context = orchestratorCtx ? { ...initialContext, ...orchestratorCtx } : initialContext;
+          let isBreakingContextBuilding = false;
 
           for (const onContext of beforeCallbacks.context) {
             const afterHookResult = await onContext({
               context,
               extendContext: extension => {
                 context = { ...context, ...extension };
               },
+              breakContextBuilding: () => {
```

Comment:
Do we need a general `shortCircuite` API from all steps? 🤔

Reply:
execute has setExecutionResultAndStopExecution but it also has drawbacks :/ https://github.com/dotansimha/envelop/issues/1156

I think in the future we can add them for all phases 🤔